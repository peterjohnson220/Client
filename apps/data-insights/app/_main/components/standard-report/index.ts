export * from './standard-report.component';
