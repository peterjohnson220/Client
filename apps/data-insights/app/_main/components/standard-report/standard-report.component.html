<div class="standard-report d-flex flex-column">
  <div class="preview d-flex justify-content-center align-items-center flex-grow-1"
  (mouseenter)="handleMouseOverWorkbookContainer()"
  (mouseleave)="handleMouseLeaveWorkbookContainer()">
    <div class="w-100">
      <a [attr.data-qa-id]="'lnk-openWorkbook'" [routerLink]="[workbook.SourceUrl,workbook.WorkbookId]"
        [queryParams]="{showTabs: workbook.ShowTabs, title: workbook.WorkbookName}">
        <img class="img-fluid" [src]="workbook.Thumbnail" />

        <div class="d-flex img-overlay" [ngClass]="displayOverlay ? 'displayed' : ''">
          <div class="d-flex justify-content-center align-items-center workbook-description">
            {{workbook.WorkbookDescription}}
          </div>
        </div>
      </a>
    </div>
  </div>
  <div class="type mt-3 mb-2">
    Payfactors Reports
  </div>
  <div class="name" [title]="workbook.WorkbookName">
    {{workbook.WorkbookName}}
  </div>
</div>


