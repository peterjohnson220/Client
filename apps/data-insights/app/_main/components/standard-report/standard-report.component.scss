@import 'colors';
@import 'common/common';
@import 'data-insights-variables';

.standard-report {
  padding: 1rem 1rem 3rem 1rem;
  position: relative;
  height: 350px;

  .preview {
    margin-bottom: 1rem;
    cursor: pointer;
    min-height: 1px;

    .img-fluid {
      width: 100%;
    }
  }

  .type {
    @extend %label;
    border-bottom: 1px solid $border-grey;
  }

  .name {
    font-size: 1.25rem;
    @extend %ellipsis-overflowing-text;
  }
}
