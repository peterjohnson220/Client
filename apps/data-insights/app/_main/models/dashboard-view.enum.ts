export enum DashboardView {
  All = 'All Dashboards',
  Favorites = 'Favorites',
  Views = 'All Views'
}
