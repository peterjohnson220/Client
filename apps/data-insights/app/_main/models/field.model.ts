export interface Field {
  EntityId: number;
  Entity: string;
  EntitySourceName: string;
  DataElementId: number;
  SourceName: string;
  DisplayName: string;
  KendoGridField?: string;
  DataType?: FieldDataType;
  IsSelected?: boolean;
  IsSortable: boolean;
  Order?: number;
  DataElementOrder?: number;
  FormulaId?: number;
  FieldType: FieldType;
  Format?: string;
  IsEditable?: boolean;
  Formula?: string;
  FormulaName?: string;
}

export interface FieldListItem {
  Entity: string;
  IsSelected: boolean;
  DisplayName: string;
  FieldListItemId: string;
}

export enum FieldDataType {
  Date = 'Date',
  Int = 'int',
  Float = 'float',
  String = 'String',
  LongString = 'longString',
  Bit = 'bit'
}

export enum FieldType {
  DataElement = 'DataElement',
  Formula = 'Formula'
}

export function generateMockField(): Field {
  return {
    DataElementId: 1,
    EntityId: 1,
    Entity: 'Jobs',
    DataType: FieldDataType.String,
    DisplayName: 'Job Title',
    EntitySourceName: 'CompanyJobs',
    IsSelected: false,
    Order: 1,
    SourceName: 'Job_Title',
    KendoGridField: 'CompanyJobs.Job_Title',
    IsSortable: true,
    FieldType: FieldType.DataElement,
    Format: 'N1'
  };
}
