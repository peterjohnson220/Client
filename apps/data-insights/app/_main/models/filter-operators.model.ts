export interface FilterOperator {
  Name: string;
  Value: string;
}

export const Equals: FilterOperator = {
  Name: 'equals',
  Value: 'in'
};

export const Between: FilterOperator = {
  Name: 'between',
  Value: 'between'
};

export const GreaterThan: FilterOperator = {
  Name: 'greater than',
  Value: '>'
};

export const GreaterThanOrEqual: FilterOperator = {
  Name: 'greater than or equal',
  Value: '>='
};

export const LessThan: FilterOperator = {
  Name: 'less than',
  Value: '<'
};

export const LessThanOrEqual: FilterOperator = {
  Name: 'less than or equal',
  Value: '<='
};

export const IsBefore: FilterOperator = {
  Name: 'is before',
  Value: '<'
};

export const IsAfter: FilterOperator = {
  Name: 'is after',
  Value: 'isafterdate'
};

export const Is: FilterOperator = {
  Name: 'is',
  Value: 'isdate'
};

export const Contains: FilterOperator = {
  Name: 'contains',
  Value: 'contains'
};
