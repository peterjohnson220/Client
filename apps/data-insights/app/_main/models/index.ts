export * from './dashboard-view.enum';
export * from './workbook.model';
export * from './save-workbook-tag-obj.model';
export * from './report-view-types.enum';
export * from './view.model';
export * from './entity.model';
export * from './save-user-workbook-modal-data.model';
export * from './user-data-view.model';
export * from './report-types.enum';
export * from './field.model';
export * from './save-workbook-mode.enum';
export * from './save-report-order-data.model';
