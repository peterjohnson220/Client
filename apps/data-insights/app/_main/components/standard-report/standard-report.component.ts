import { Component, Input } from '@angular/core';

import { Workbook } from '../../models';

@Component({
  selector: 'pf-standard-report',
  templateUrl: './standard-report.component.html',
  styleUrls: ['./standard-report.component.scss']
})
export class StandardReportComponent {
  @Input() workbook: Workbook;

  constructor() {}
}
