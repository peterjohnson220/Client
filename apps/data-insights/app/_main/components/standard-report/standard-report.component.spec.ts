import { ComponentFixture, TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';

import { StandardReportComponent } from './standard-report.component';
import { generateMockWorkbook } from '../../models';

describe('Data Insights - Standard Report Component', () => {
  let instance: StandardReportComponent;
  let fixture: ComponentFixture<StandardReportComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [ StandardReportComponent ],
      schemas: [NO_ERRORS_SCHEMA]
    });

    fixture = TestBed.createComponent(StandardReportComponent);
    instance = fixture.componentInstance;

    fixture.detectChanges();
  });

  it('should set displayOverlay to true when handling mouse over workbook with description', () => {
    instance.workbook = generateMockWorkbook();

    instance.handleMouseOverWorkbookContainer();

    expect(instance.displayOverlay).toEqual(true);
  });

  it('should NOT set displayOverlay to true when handling mouse over workbook with NO description', () => {
    instance.workbook = { ...generateMockWorkbook(), WorkbookDescription: null };
    instance.displayOverlay = false;

    instance.handleMouseOverWorkbookContainer();

    expect(instance.displayOverlay).toEqual(false);
  });

  it('should set displayOverlay to false when handling mouse leave workbook with description', () => {
    instance.workbook = generateMockWorkbook();

    instance.handleMouseLeaveWorkbookContainer();

    expect(instance.displayOverlay).toEqual(false);
  });
});
