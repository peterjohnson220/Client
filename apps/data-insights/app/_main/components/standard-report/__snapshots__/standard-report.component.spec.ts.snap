// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Insights - Standard Report Component should match snapshot 1`] = `
<pf-standard-report
  router={[Function Object]}
>
  
</pf-standard-report>
`;
